src/ranged.cpp

```diff
@@ -1416,20 +1421,28 @@ static int rand_or_max( bool random, int max )
 // utility functions for projectile_attack
 double player::get_weapon_dispersion( const item *weapon, bool random ) const
 {
-    if( weapon->is_gun() && weapon->is_in_auxiliary_mode() ) {
-        const auto gunmod = weapon->gunmod_current();
-        if( gunmod != nullptr ) {
-            return get_weapon_dispersion( gunmod, random );
-        }
-    }
-
     double dispersion = 0.; // Measured in quarter-degrees.
     dispersion += skill_dispersion( *weapon, random );
 
+    if( is_driving() ) {
+        // get volume of gun (or for auxiliary gunmods the parent gun)
+        const item *parent = has_item( *weapon ) ? find_parent( *weapon ) : nullptr;
+        int vol = parent ? parent->volume() : weapon->volume();
+
+        ///\EFFECT_DRIVING reduces the inaccuracy penalty when using guns whilst driving
+        dispersion += std::max( vol - get_skill_level( skill_driving ), 1 ) * 20;
+    }
+
     dispersion += rand_or_max( random, ranged_dex_mod() );
     dispersion += rand_or_max( random, ranged_per_mod() );
 
-    dispersion += rand_or_max( random, 3 * (encumb(bp_arm_l) + encumb(bp_arm_r)));
+    // pistols select the least encumbered hand
+    if( weapon->gun_skill() == skill_pistol ) {
+        dispersion += rand_or_max( random, 3 * std::min( encumb( bp_arm_l ), encumb( bp_arm_r ) ) );
```

Comment:
> What behavior do you have in mind?

For pistols, the formula is `3 * min(x, y)`, for everything else `3 * (x+y)`. When `x==y`, pistols have 50% of the dispersion from encumbrance. Giving pistols `6 * min(x, y)` should make those two formulas equivalent for equally encumbered arms, but favoring pistols when they aren't.

> I was actually intending for that to be a feature of pistols.

Light SMGs should also have that. And some "strange" weapons, like slings.

Overly-modified, huge or powerful pistols shouldn't have this ability. Things like the rivtech pistol-shotgun, rivtech automagnum, desert eagle etc.

If it was a function of volume alone, it would have to be `volume <= 2` to make sense. It would probably be better off as a per-item flag, with stock/grip mods removing it.

Though using pistols one-handed should be limited to snapshots. It shouldn't allow bringing total dispersion below two handed use.